File: packages/core/src/document-utils.ts
import type { DocumentType } from '@qtscout/types/document'
import { DOCUMENT_TYPE_PREFIXES } from '@qtscout/types/document'

export function formatDocumentIdentifier(type: DocumentType, number: number, year: number | null): string {
  const prefix = DOCUMENT_TYPE_PREFIXES[type]
  const num = number.toString().padStart(3, '0')
  if (type === 'ORDEM_SERVICO') {
    return `${prefix}-${num}`
  }
  return `${prefix}-${num}/${year}`
}

File: api/app/core/document_utils.py
"""Mirrors `@qtscout/core/document-utils` + DOCUMENT_TYPE_PREFIXES."""

from app.models.enums import DocumentType

DOCUMENT_TYPE_PREFIXES: dict[DocumentType, str] = {
    DocumentType.OFICIO: "OF",
    DocumentType.CIRCULAR: "CI",
    DocumentType.ORDEM_SERVICO: "OS",
}


def format_document_identifier(type_: DocumentType, number: int, year: int | None) -> str:
    prefix = DOCUMENT_TYPE_PREFIXES[type_]
    num = str(number).zfill(3)
    if type_ == DocumentType.ORDEM_SERVICO:
        return f"{prefix}-{num}"
    return f"{prefix}-{num}/{year}"
